app/files.js

```diff
@@ -312,6 +375,54 @@ const roomArchived = {
   }
 }
 
+const shareRoles = {
+  key: "shareRoles",
+  noun: "Room",
+  display: {
+    label: "Get Roles",
+    description: "Get roles for share by room id.",
+    hidden: true
+  },
+  operation: {
+    /**
+     * @param {ZObject} z
+     * @param {Bundle<SessionAuthenticationData, ShareRolesFields>} bundle
+     * @returns {Promise<RoleData[]>}
+     */
+    async perform(z, bundle) {
+      const client = new Client(bundle.authData.baseUrl, z.request)
+      const files = new FilesService(client)
+      const room = await files.roomInfo(bundle.inputData.id)
+      if (isPublicRoom(room.roomType)) {
+        return [
+          { id: 2, name: "Viewer" },
+          { id: 10, name: "Editor" }
+        ]
+      }
+      if (isCustomRoom(room.roomType)) {
+        return [
+          { id: 9, name: "Room admin" },
+          { id: 11, name: "Power user" }
+        ]
+      }
+      return []
+      /**
+       * @param {number} type
+       */
+      function isPublicRoom(type) {
+        return type === 5 // or another number, idk
```

Comment:
My comment was intended solely for the purpose of providing clarity and should not have been included in the source code.

Reply:
> My comment was intended solely for the purpose of providing clarity and should not have been included in the source code.

Sorry, copied it without noticing.